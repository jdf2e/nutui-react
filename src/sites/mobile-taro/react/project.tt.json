{
  "miniprogramRoot": "./",
  "projectname": "first",
  "appid": "testAppId",
  "setting": {
    "es6": false,
    "minified": false
  }
}
