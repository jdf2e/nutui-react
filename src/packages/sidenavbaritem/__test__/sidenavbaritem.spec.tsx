import React from 'react'

import { render, fireEvent } from '@testing-library/react'
import '@testing-library/jest-dom'

import SideNavBarItem from '@/packages/sidenavbaritem'

test('should match snapshot', () => {
  const { asFragment } = render(
    <SideNavBarItem
      data-testid="sideNavBarItem-click"
      title="一级内容1"
      ikey="1-01"
    />
  )
  expect(asFragment()).toMatchSnapshot()
})
test('should emit click event', async () => {
  const click = jest.fn(({ title, ikey }) => {
    console.log('click>>>>', title, ikey)
  })
  const { getByTestId } = render(
    <>
      <SideNavBarItem
        data-testid="sideNavBarItem-click"
        title="一级内容1"
        ikey="1-01"
        click={click}
      />
    </>
  )
  fireEvent.click(getByTestId('sideNavBarItem-click'))
  expect(click).toBeCalled()
})
