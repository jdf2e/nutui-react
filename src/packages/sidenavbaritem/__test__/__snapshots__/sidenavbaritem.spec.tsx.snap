// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="nut-subsidenavbar-item nut-subsidenavbar-border-bt"
    data-testid="sideNavBarItem-click"
  >
    一级内容1
  </div>
</DocumentFragment>
`;
